optuna/visualization.py

```diff
@@ -92,6 +99,164 @@ def _get_intermediate_plot(study):
     return figure
 
 
+def plot_contour(study, params=[]):
+    # type: (Study, List[str]) -> None
+    """Plot the parameter relationship as contour plot in a study.
+
+        Note that, If a parameter contains missing values, a trial with missing values is not
+        plotted.
+
+    Example:
+
+        The following code snippet shows how to plot the parameter relationship as contour plot
+        inside Jupyter Notebook.
+
+        .. code::
+
+            import optuna
+
+            def objective(trial):
+                ...
+
+            study = optuna.create_study()
+            study.optimize(objective, n_trials=100)
+
+            optuna.visualization.plot_contour(study, params=['param_a', 'param_b'])
+
+    Args:
+        study:
+            A :class:`~optuna.study.Study` object whose trials are plotted for their objective
+            values.
+        params:
+            Parameter list to visualize. The default is all parameters.
+    """
+
+    _check_plotly_availability()
+    init_notebook_mode(connected=True)
+    figure = _get_contour_plot(study, params)
+    figure.show()
+
+
+def _get_contour_plot(study, params=[]):
+    # type: (Study, List[str]) -> Figure
+    layout = go.Layout(
+        title='Contour Plot',
+    )
+
+    trials = [trial for trial in study.trials if trial.state == TrialState.COMPLETE]
+
+    if len(trials) == 0:
+        logger.warning('Your study does not have any completed trials.')
+        return go.Figure(data=[], layout=layout)
+
+    all_params = {p_name for t in trials for p_name in t.params.keys()}
+    if len(params) == 0:
+        sorted_params = sorted(list(all_params))
+    elif len(params) == 1:
+        logger.warning('The length of params must be greater than 1.')
+        return go.Figure(data=[], layout=layout)
+    else:
+        for input_p_name in params:
+            if input_p_name not in all_params:
+                logger.warning('Parameter {} does not exist in your study.'.format(input_p_name))
```

Comment:
In my opinion, we should raise an error instead of warning. Otherwise, the user might need additional effort to specify the cause of the empty figure.

Reply:
I agree with @g-votte.